Source/Core/CoplanarPolygonGeometryLibrary.js

```diff
@@ -0,0 +1,131 @@
+define([
+        './defined',
+        './Cartesian2',
+        './Cartesian3',
+        './Check',
+        './IntersectionTests',
+        './Math',
+        './Matrix3',
+        './OrientedBoundingBox',
+        './Plane',
+        './Ray'
+    ], function(
+        defined,
+        Cartesian2,
+        Cartesian3,
+        Check,
+        IntersectionTests,
+        CesiumMath,
+        Matrix3,
+        OrientedBoundingBox,
+        Plane,
+        Ray
+    ) {
+    'use strict';
+
+    /**
+     * @private
+     */
+    var PolygonGeometryLibrary = {};
+
+    var scratchIntersectionPoint = new Cartesian3();
+    var scratchXAxis = new Cartesian3();
+    var scratchYAxis = new Cartesian3();
+    var scratchZAxis = new Cartesian3();
+    var scratchOrigin = new Cartesian3();
+    var scratchNormal = new Cartesian3();
+    var scratchRay = new Ray();
+    var scratchPlane = new Plane(Cartesian3.UNIT_X, 0);
+    var obbScratch = new OrientedBoundingBox();
+
+    // call after removeDuplicates
+    PolygonGeometryLibrary.projectTo2D = function(positions, positionsResult, normalResult, tangentResult, bitangentResult) {
+        //>>includeStart('debug', pragmas.debug);
+        Check.defined('positions', positions);
+        Check.defined('positionsResult', positionsResult);
+        //>>includeEnd('debug');
+
+        var obb = OrientedBoundingBox.fromPoints(positions, obbScratch);
+        var halfAxes = obb.halfAxes;
+        var xAxis = Matrix3.getColumn(halfAxes, 0, scratchXAxis);
+        var yAxis = Matrix3.getColumn(halfAxes, 1, scratchYAxis);
+        var zAxis = Matrix3.getColumn(halfAxes, 2, scratchZAxis);
+
+        var xMag = Cartesian3.magnitude(xAxis);
+        var yMag = Cartesian3.magnitude(yAxis);
+        var zMag = Cartesian3.magnitude(zAxis);
+        var min = Math.min(xMag, yMag, zMag);
+
+        var i;
+        // If all the points are on a line, just remove one of the zero dimensions
+        if ((xMag === 0 && (yMag === 0 || zMag === 0)) || (yMag === 0 && zMag === 0)) {
+            return;
+        }
+        var center = obb.center;
+        var planeXAxis;
+        var planeYAxis;
+        var origin = Cartesian3.clone(center, scratchOrigin);
+        var normal;
+        if (min === xMag) {
```

Comment:
This can probably be more simple. Once you find the vectors with the minimum and maximum length:
``javascript
z = cross(x, y); // normal
y = cross(z, x);
// normalize x, y, and z
```
It doesn't matter if you use z or -z for the normal. It also doesn't matter if you use +/-x or +/-y as long as they are orthogonal in the plane.